controllers/goharbor/exporter/deployments.go

```diff
@@ -0,0 +1,241 @@
+		})
+
+		volumes = append(volumes, corev1.Volume{
+			Name: InternalCertificatesVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: exporter.Spec.TLS.CertificateRef,
+				},
+			},
+		})
+	} else {
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+			Name:      InternalCertificatesVolumeName,
+			MountPath: InternalCertificateAuthorityDirectory,
+		})
+
+		volumes = append(volumes, corev1.Volume{
+			Name: InternalCertificatesVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		})
+	}
+
+	httpGET := &corev1.HTTPGetAction{
+		Path:   HealthPath,
+		Port:   intstr.FromString(harbormetav1.ExporterMetricsPortName),
+		Scheme: corev1.URISchemeHTTP,
+	}
+
+	deploy := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        name,
+			Namespace:   namespace,
+			Annotations: version.NewVersionAnnotations(exporter.Annotations),
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					r.Label("name"):      name,
+					r.Label("namespace"): namespace,
+				},
+			},
+			Replicas: exporter.Spec.Replicas,
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						r.Label("name"):      name,
+						r.Label("namespace"): namespace,
+					},
+				},
+				Spec: corev1.PodSpec{
+					NodeSelector:                 exporter.Spec.NodeSelector,
+					AutomountServiceAccountToken: &varFalse,
+					Volumes:                      volumes,
+					SecurityContext: &corev1.PodSecurityContext{
+						FSGroup:    &fsGroup,
+						RunAsGroup: &runAsGroup,
+						RunAsUser:  &runAsUser,
+					},
+					Containers: []corev1.Container{{
+						Name:  controllers.Exporter.String(),
+						Image: image,
+						Args: []string{
+							"-log-level",
+							string(exporter.Spec.Log.Level),
+						},
+						Ports: []corev1.ContainerPort{{
+							Name:          harbormetav1.ExporterMetricsPortName,
+							ContainerPort: exporter.Spec.Port,
+							Protocol:      corev1.ProtocolTCP,
+						}},
+
+						Env: envs,
+
+						VolumeMounts: volumeMounts,
+
+						LivenessProbe: &corev1.Probe{
+							Handler: corev1.Handler{
+								HTTPGet: httpGET,
+							},
+						},
+						ReadinessProbe: &corev1.Probe{
+							Handler: corev1.Handler{
+								HTTPGet: httpGET,
+							},
+						},
+					}},
+				},
+			},
+		},
+	}
+
+	exporter.Spec.ComponentSpec.ApplyToDeployment(deploy)
+
+	return deploy, nil
+}
+
+func parseServiceInfo(coreURL string) (scheme string, host string, port string, err error) {
+	u, err := url.Parse(coreURL)
+	if err != nil {
+		return "", "", "", err
+	}
+
+	scheme = strings.ToLower(u.Scheme)
+	if scheme == "http" {
+		port = "80"
+	} else if scheme == "https" {
+		port = "443"
+	}
+
+	parts := strings.Split(u.Host, ":")
+	host = parts[0]
+
+	if len(parts) == 2 { // nolint:gomnd
+		port = parts[1]
+	}
```

Comment:
I suggest something that imo is more readable (no split and array index):

```suggestion
	scheme = u.Scheme
	host = u.Hostname()
	port = u.Port()
	
	if port == "" {
		switch strings.ToLower(scheme) {
			case "http":
				port = "80"
			default:
				port = "443"
		}
	}
```